spec/decorators/search_result_collection_decorator_spec.rb

```diff
@@ -1,212 +1,137 @@
+      {
+        results: [result],
+        total_results: total_results,
+        page: page,
+        number_of_pages: number_of_pages,
+        per_page: per_page,
+        query: query
+      }
+    )
+  end
+  let(:total_results) { 1 }
+  let(:page) { 0 }
+  let(:number_of_pages) { 1 }
+  let(:per_page) { 10 }
+  let(:query) { 'budget planner' }
 
   it "decorates the collection's items with SearchResultDecorator" do
     expect(subject.first).to be_a SearchResultDecorator
   end
 
   describe '#page' do
-    let(:number_of_pages) { 10 }
+    let(:page) { 0 }
 
-    before do
-      allow(subject).to receive(:number_of_pages) { number_of_pages }
-      allow(result_collection).to receive(:page) { page }
-    end
-
-    context 'when the requested page number is less than the total number of pages' do
-      let(:page) { 5 }
-
-      it 'returns the original requested page number' do
-        expect(subject.page).to eq page
-      end
-    end
-
-    context 'when the requested page number is more than the total number of pages' do
-      let(:page) { 15 }
-
-      it 'returns the total number of pages' do
-        expect(subject.page).to eq number_of_pages
-      end
-    end
-  end
-
-  describe '#total_results' do
-    let(:result_limit) { 50 }
-
-    before do
-      allow(subject).to receive(:result_limit) { result_limit }
-      allow(result_collection).to receive(:total_results) { total_results }
-    end
-
-    context 'when the total number of results is less than the result limit' do
-      let(:total_results) { 40 }
-
-      it 'returns the actual number of results' do
-        expect(subject.total_results).to eq total_results
-      end
-    end
-
-    context 'when the total number of results is more than the result limit' do
-      let(:total_results) { 100 }
-
-      it 'returns the result limit' do
-        expect(subject.total_results).to eq result_limit
-      end
+    it 'added one more to the page number' do
+      expect(subject.page).to be(1)
     end
   end
 
   context 'with 5 pages of 10 results' do
-    before do
-      result_collection.per_page      = 10
-      result_collection.total_results = 50
-    end
+    let(:per_page) { 10 }
+    let(:total_results) { 50 }
 
     describe '#first_page?' do
       subject { decorator.first_page? }
 
-      context 'when the current page is 1' do
-        before { result_collection.page = 1 }
+      context 'when first page' do
+        let(:page) { 0 }
+
         it { is_expected.to be_truthy }
       end
 
-      context 'when the current page is 3' do
-        before { result_collection.page = 3 }
+      context 'when third page' do
+        let(:page) { 2 }
+
         it { is_expected.to be_falsey }
       end
 
-      context 'when the current page is 5' do
-        before { result_collection.page = 5 }
+      context 'when fifth page' do
+        let(:page) { 4 }
+
         it { is_expected.to be_falsey }
       end
     end
 
     describe '#last_page?' do
```

Comment:
given we have deleted the feature for last page and no longer limited to only 5 pages are these tests still relevant?

Reply:
Yes because if the last page is 10 for example it should return if I am on the last page and should not display the next button.

Reply:
ok so taking your comment above I'm happy for another PR to be opened for a feature test around this then :)